Traceback (most recent call last):
  File "/Users/andra/Library/Python/3.9/lib/python/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/andra/Library/Python/3.9/lib/python/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/andra/Library/Python/3.9/lib/python/site-packages/jupyter_core/utils/__init__.py", line 166, in wrapped
    return loop.run_until_complete(inner)
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/Users/andra/Library/Python/3.9/lib/python/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/Users/andra/Library/Python/3.9/lib/python/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/andra/Library/Python/3.9/lib/python/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
!pip install wikidataintegrator
from wikidataintegrator import wdi_core
from IPython.display import Markdown, display

# Set up the SPARQL endpoint URL
sparql = "https://query.wikidata.org/sparql"

# Set up the SPARQL query
query = """
SELECT ?publication ?publicationLabel ?doi
WHERE
{
  ?publication wdt:P50 wd:Q19845625.
  ?publication wdt:P31 wd:Q13442814.
  OPTIONAL { ?publication wdt:P356 ?doi. }
  SERVICE wikibase:label { bd:serviceParam wikibase:language "en". }
}
"""

results = wdi_core.WDItemEngine.execute_sparql_query(query, as_dataframe=False, endpoint=sparql)

# Print the list of publications with their DOI and QID

md = ""
for result in results["results"]["bindings"]:
    publication_name = result["publicationLabel"]["value"]
    publication_doi = result["doi"]["value"] if "doi" in result else "N/A"
    publication_qid = result["publication"]["value"].split("/")[-1]
    if publication_doi and publication_qid:
        md += f"* {publication_name} (<a href='https://doi.org/{publication_doi}' target='_blank'><img src='240px-DOI_logo.svg.png' alt='DOI' height='20'></a>, <a href='https://www.wikidata.org/wiki/{publication_qid}' target='_blank'><img src='320px-Wikidata-logo.svg' alt='Wikidata'></a>)\n"
    elif publication_qid:
        md += f"* {publication_name} (<a href='https://www.wikidata.org/wiki/{publication_qid}' target='_blank'><img src='320px-Wikidata-logo.svg' alt='Wikidata'></a>)\n"
    else:
        md += f"* {publication_name}\n"
display(Markdown(md))
------------------

----- stdout -----
zsh:1: command not found: pip
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mModuleNotFoundError[0m                       Traceback (most recent call last)
Cell [0;32mIn[1], line 2[0m
[1;32m      1[0m get_ipython()[38;5;241m.[39msystem([38;5;124m'[39m[38;5;124mpip install wikidataintegrator[39m[38;5;124m'[39m)
[0;32m----> 2[0m [38;5;28;01mfrom[39;00m [38;5;21;01mwikidataintegrator[39;00m [38;5;28;01mimport[39;00m wdi_core
[1;32m      3[0m [38;5;28;01mfrom[39;00m [38;5;21;01mIPython[39;00m[38;5;21;01m.[39;00m[38;5;21;01mdisplay[39;00m [38;5;28;01mimport[39;00m Markdown, display
[1;32m      5[0m [38;5;66;03m# Set up the SPARQL endpoint URL[39;00m

[0;31mModuleNotFoundError[0m: No module named 'wikidataintegrator'

